src/html5-qrcode.js

```diff
@@ -797,156 +797,60 @@ class Html5Qrcode {
         };
     }
 
-    _possiblyInsertShadingElement(element, height, qrRegion) {
-        if (qrRegion.x == 0 && qrRegion.y == 0) {
-            // No shading
-            return;
-        }
-
-        const shaders = {};
-        shaders[Html5Qrcode.SHADED_LEFT] = this._createShadedElement(
-            height, qrRegion, Html5Qrcode.SHADED_LEFT);
-        shaders[Html5Qrcode.SHADED_RIGHT] = this._createShadedElement(
-            height, qrRegion, Html5Qrcode.SHADED_RIGHT);
-        shaders[Html5Qrcode.SHADED_TOP] = this._createShadedElement(
-            height, qrRegion, Html5Qrcode.SHADED_TOP);
-        shaders[Html5Qrcode.SHADED_BOTTOM] = this._createShadedElement(
-            height, qrRegion, Html5Qrcode.SHADED_BOTTOM);
-
-        Object.keys(shaders).forEach(key => element.append(shaders[key]));
-
-        if (qrRegion.x < 10 || qrRegion.y < 10) {
-            this.hasBorderShaders = false;
-        } else {
-            Object.keys(shaders).forEach(key =>
-                this._insertShaderBorders(shaders[key], qrRegion, key));
-            this.hasBorderShaders = true;
-        }
+    _possiblyInsertShadingElement(element, width, height, qrboxSize) {
+      if((width - qrboxSize) < 1 || (height - qrboxSize) < 1){
+        return;
+      }
+      const elem = document.createElement('div');
+      elem.style.position = "absolute";
+      elem.style.borderLeft = `${(width-qrboxSize)/2}px solid #0000007a`;
+      elem.style.borderRight = `${(width-qrboxSize)/2}px solid #0000007a`;
+      elem.style.borderTop = `${(height-qrboxSize)/2}px solid #0000007a`;
+      elem.style.borderBottom = `${(height-qrboxSize)/2}px solid #0000007a`;
+      elem.style.boxSizing = "border-box";
+      elem.style.top = "0px";
+      elem.style.bottom = "0px";
+      elem.style.left = "0px";
+      elem.style.right = "0px";
```

Comment:
isn't `top` and `left` enough here? Does setting all the positions add value?

Reply:
top and left are used to define the start position. To get a div that span completely above the relative div either the width and height needs to be supplied or the bottom and right point needs to be defined. I chose the second option so that I don't need to know the height and width at this point.